src/main/java/de/rondiplomatico/nds/floodFill.java

```diff
@@ -0,0 +1,271 @@
+package de.rondiplomatico.nds;
+
+import java.awt.Point;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Utility class for flood fill algorithm applied to marked tiles deduced from polygon points.
+ * 
+ * @author Andreas Hessenthaler
+ * @since 13.02.2020
+ */
+public class floodFill {
+	int[][] ff;
+	int minX 					= Integer.MAX_VALUE;
+	int maxX 					= Integer.MIN_VALUE;
+	int minY 					= Integer.MAX_VALUE;
+	int maxY 					= Integer.MIN_VALUE;
+	int dimX 					= -1;
+	int dimY 					= -1;
+	int bg 						= 0;
+	int bound 					= 1;
+	int fill 					= bound;
+	int startAtX 				= 0;
+	int startAtY 				= 0;
+	
+	public floodFill(int[][] f, int a, int b, int c, int d) {
+		ff 						= f;
+		minX 					= a;
+		maxX 					= b;
+		minY 					= c;
+		maxY 					= d;
+		dimX 					= maxX - minX + 1;
+		dimY 					= maxY - minY + 1;
+		setStartingPoint();
+	}
+	
+	/*
+	 * Four-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right
+			recursiveFloodFill(x,   y+1); // top
+			recursiveFloodFill(x-1, y  ); // left
+			recursiveFloodFill(x,   y-1); // bottom
+		}
+		return;
+	}
+
+	/*
+	 * Eight-way recursive flood fill algorithm
+	 * 
+	 * @param x
+	 * 			the tile x coordinate
+	 * @param y
+	 * 			the tile y coordinate
+	 */
+	public void recursiveFloodFill8(int x, int y) {
+		if ((x < 0) || (y < 0) || (x > dimX-1 || (y > dimY-1))) {
+			return;
+		}
+		if (ff[x][y] == bg) {
+			ff[x][y] 			= fill;
+			recursiveFloodFill(x+1, y  ); // right

```

Comment:
Wouldn't those invocations all need to be `recursiveFloodFill8` ?

Reply:
Yes - fixed.